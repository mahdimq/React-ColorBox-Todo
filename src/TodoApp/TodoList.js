import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

function TodoList() {
	const INITIAL_STATE = [{ id: uuid(), todo: '' }];

	const [todos, setTodos] = useState(INITIAL_STATE);

	const addTodo = (newTodo) => {
		setTodos((todos) => [...todos, { id: uuid(), ...newTodo }]);
	};

	const handleDelete = (id) => {
		setTodos((todos) => todos.filter((todo) => todo.id !== id));
	};

	return (
		<div>
			<NewTodoForm addTodo={addTodo} />
			{todos.map((items) => (
				<Todo todo={items.todo} key={items.id} id={items.id} deleteTodo={handleDelete} />
			))}
		</div>
	);
}

export default TodoList;
