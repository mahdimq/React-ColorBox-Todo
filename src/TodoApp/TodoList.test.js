import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

// SMOKE TEST
it('renders without crashing', function () {
	render(<TodoList />);
});
// SNAPSHOT TEST
it('matches snapshot', function () {
	const { asFragment } = render(<TodoList />);
	expect(asFragment()).toMatchSnapshot();
});

it('can add a new todo', function () {
	const { getByLabelText, queryByText } = render(<TodoList />);

	expect(queryByText('X')).not.toBeInTheDocument();

	const input = getByLabelText('Enter Todo');
	const submitBtn = queryByText('Add todo');

	// fill out the form
	fireEvent.change(input, { target: { value: 'Iron Clothes' } });
	fireEvent.click(submitBtn);

	// todo exists!
	expect(queryByText('Iron Clothes')).toBeInTheDocument();
});

it('can remove a todo', function () {
	const { getByLabelText, queryByText, getByText } = render(<TodoList />);

	const input = getByLabelText('Enter Todo');
	const submitBtn = getByText('Add todo');

	// fill out the form
	fireEvent.change(input, { target: { value: 'Wash dishes' } });
	fireEvent.click(submitBtn);
	const removeBtn = getByText('X');

	expect(queryByText('Wash dishes')).toBeInTheDocument();
	fireEvent.click(removeBtn);
	// todo does not exist!
	expect(queryByText('Wash dishes')).not.toBeInTheDocument();
});
