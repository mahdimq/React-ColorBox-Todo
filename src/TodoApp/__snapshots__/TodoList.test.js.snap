// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="todo"
      >
        Enter Todo
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Wash car..."
        type="text"
        value=""
      />
      <button>
        Add todo
      </button>
    </form>
    <div>
      <span />
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
